### PR TITLE
2.0: rewrite polar/ by ts

### PR DESCRIPTION
**Pull request by xile611**

### CODE REVIEW

File: src/polar/Pie.tsx [hunk 1]
```diff
@@ -1,99 +1,110 @@
 /**
  * @fileOverview Render sectors of a pie
  */
-import React, { PureComponent } from 'react';
-import PropTypes from 'prop-types';
+import React, { PureComponent, FunctionComponent, Component, ReactNode, ReactElement } from 'react';
```
Comment (sainthkh) [inline src/polar/Pie.tsx]:
```suggestion
import React, { PureComponent, ReactElement } from 'react';
```

File: src/util/types.ts [hunk 2]
```diff
@@ -27,9 +31,31 @@ export type PresentationAttributesWithProps<P, T> = AriaAttributes &
...
+  TIME = 'time', 
+  BAND = 'band', 
+  POINT = 'point',
+  ORDINAL = 'ordinal', 
+  QUANTILE = 'quantile', 
+  QUANTIZE = 'quantize', 
+  UTC = 'utc', 
+  SEQUENTIAL = 'sequential', 
+  THRESHOLD = 'threshold',
+};
```
Comment (sainthkh) [inline src/util/types.ts]:
```ts
export type ScaleType = 'auto' | 'linear' | 'pow' | ...
```
Isn't above is shorter and better than enum type? Will we use it like `scaleType == ScaleType.AUTO`?

Comment (xile611) [reply src/util/types.ts]: Good suggestion! I'll change this.

**Comment by Philipp91** [comment]
The type `ReactElement<SVGElement> | ((props: any) => SVGElement)` occurs several times in this PR. It doesn't work for me in practice. I'm trying to feed a function component to `Pie.activeShape`, which TypeScript refuses saying that `Type '(props: ...) => JSX.Element' is not assignable to type '(props: any) => SVGElement'.`

Should it be `ReactElement<SVGElement> | ((props: any) => ReactElement<SVGElement>)` instead? I.e. "either you have rendered the element already or you supply a function that will".

**Pull request merged**